sound/soc/sof/pcm.c

```diff
@@ -687,6 +687,37 @@ static int sof_pcm_dai_link_fixup(struct snd_soc_pcm_runtime *rtd,
 	return 0;
 }
 
+static void sof_set_longname(struct snd_soc_component *component)
+{
+	struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(component);
+	struct snd_sof_pdata *pdata = sdev->pdata;
+	char flavour[80];
+	int pos = 0;
+	int i;
+
+	pos += sprintf(flavour, "%s", "SOF");
+	pos += sprintf(flavour + pos, "-%02d%02d%02d-",
+		       sdev->fw_version.major,
+		       sdev->fw_version.minor,
+		       sdev->fw_version.micro);
+
+	for (i = 0; pdata->tplg_filename[i]; i++) {
+		if (pdata->tplg_filename[i] == '.')
+			break;
+		if (isalnum(pdata->tplg_filename[i]))
+			flavour[pos++] = pdata->tplg_filename[i];
+	}
+
+	sprintf(flavour + pos, "-%02d%02d%02d",
+		SOF_ABI_VERSION_MAJOR(sdev->fw_version.abi_version),
+		SOF_ABI_VERSION_MINOR(sdev->fw_version.abi_version),
+		SOF_ABI_VERSION_PATCH(sdev->fw_version.abi_version));
```

Comment:
Same here. I don't think this is useful for UCM selection, at least via long cardname. This is just the version of the communication protocal with the DSP -- has not much to do with what features are supported.

Reply:
yes, I was just trying corporate everything that was discussed in alsa-devel.